src/shared/sam/cli/samCliValidationUtils.ts

```diff
@@ -0,0 +1,48 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import { notifySamCliValidation } from './samCliValidationNotification'
+import {
+    InvalidSamCliError,
+    InvalidSamCliVersionError,
+    SamCliNotFoundError,
+    SamCliValidatorResult,
+    SamCliVersionValidation
+} from './samCliValidator'
+
+export function throwIfInvalid(validationResult: SamCliValidatorResult): void {
+    if (!validationResult.samCliFound) {
+        throw new SamCliNotFoundError()
+    }
+
+    if (!validationResult.versionValidation) {
+        // This should never happen
+        throw new Error('SAM CLI detected but version validation is missing')
+    }
+
+    if (validationResult.versionValidation.validation === SamCliVersionValidation.Valid) {
+        // valid state
+        return
+    }
+
+    // Invalid version
+    throw new InvalidSamCliVersionError(validationResult.versionValidation)
+}
+
+export function throwAndNotifyIfInvalid(validationResult: SamCliValidatorResult): void {
```

Comment:
Does this need to be a separate file? I feel like this logic is specific enough to the samCliValidator that it can be added to that file instead of as a util.

Reply:
This will be called by higher level constructs, like the commands that create / deploy a SAM App. See `createNewSamApp.ts` as the first example of this (deploy will be in a subsequent task/PR)

Reply:
Offline conversation confirmed that these functions creates a circular dependency if this isn't in a separate file.